crates/emitter/src/emitter.rs

```diff
@@ -1055,34 +1066,34 @@ impl InstructionWriter {
         self.write_u8(depth_hint);
     }
 
-    pub fn goto_(&mut self, offset: i32) {
+    pub fn goto_(&mut self, offset: BytecodeOffsetDiff) {
```

Comment:
this part is generated by `crates/emitter/scripts/update_opcodes.py` file
https://github.com/mozilla-spidermonkey/jsparagus/blob/master/crates/emitter/scripts/update_opcodes.py
and these definition is created from `m-c/js/src/vm/Opcodes.h`

Then, you can filter the parameter types like the following:
https://github.com/mozilla-spidermonkey/jsparagus/blob/7346faa7aa45cf9b3873b097243402a680b98c1a/crates/emitter/scripts/update_opcodes.py#L380-L391

So, here, opcodes that has `JOF_JUMP` format's first operand should be `u32` and it should be replaced by `BytecodeOffsetDiff`,
and we should have `write_bytecode_offset_diff` method that unwraps the value and calls `write_i32`


The script can be run by `make update-opcodes-m-u`, as long as you have up-to-date `mozilla-unified` directory next to `jsparagus` clone.
or run `jsparagus_build_venv/bin/python crates/emitter/scripts/update_opcodes.py PATH_TO_M_C PATH_TO_JSPARAGUS`

Reply:
eventually, this change should be reflected to `Opcodes.h`
https://searchfox.org/mozilla-central/rev/4d2a9d5dc8f0e65807ee66e2b04c64596c643b7a/js/src/vm/Opcodes.h#2304
```
    /*
     * Jump to a 32-bit offset from the current bytecode.
     *
     * See "Jump instructions" above for details.
     *
     *   Category: Control flow
     *   Type: Jumps
     *   Operands: int32_t offset
     *   Stack: =>
     */ \
    MACRO(Goto, goto_, NULL, 5, 0, 0, JOF_JUMP) \
```

so, instead of `int32_t`, it should say `BytecodeOffsetDiff`.
so that emitter.rs will be created with better types